### PR TITLE
feat(github): invoke workflow (create workflow dispatch event) from element template

### PR DESCRIPTION
**Pull request by ev-codes**
## Description

Create Github Actions workflow dispatch event (to use Github's terms) from the OOTB element template.

New dispatch event operation inputs in Desktop Modeler

### Testing

I triggered workflows in [a private personal GitHub repo](https://github.com/ev-codes/github-connector-test/actions). If you'd like access, let me know.

#### Triggered Workflows

The following performed as expected:
- [X] without any inputs defined in request and workflow
- [X] with Workflow Inputs = null in request, but default variables in workflow
- [X] with Workflow Inputs = {} in request, but default variables in workflow
- [X] with defined inputs in request and workflow

The following changes didn't impact the validity of defined inputs requests:
- [X] Starting with "Issues" operation group and switching to "Actions"
- [X] Starting with "Issues", switching to "Branches", then switching back to "Actions"

Other:
- [x] From "Actions", switching to "Branches". "List branches" operation should still work.
    - Not filling in required values in "Actions" before switching to "Branches" will result in a `CREATE command rejected with INVALID ARGUMENT` error in the Modeler. This also happens with other operation groups (e.g., switching from "Branches" to "Issues"). I'll create a Modeler issue for this bug if it doesn't exist already.

### Notes

- No default value for the operationType dropdown. This is due to [a potential bug](https://github.com/camunda/web-modeler/issues/7649) in Desktop Modeler where one-option dropdown choices never trigger conditions (like `url`) when a default is applied.

### Relevant Links

- [API doc - Create Workflow Dispatch Event](https://docs.github.com/en/rest/actions/workflows?apiVersion=2022-11-28#create-a-workflow-dispatch-event)
- [General doc - Github Actions](https://docs.github.com/en/actions/learn-github-actions/understanding-github-actions)

## Related issues

https://github.com/camunda/connectors/issues/1074

## Next Steps (post-merge)

- Add this template to Web Modeler
- Update [Github OOTB Connector](https://docs.camunda.io/docs/components/connectors/out-of-the-box-connectors/github/) doc

### CODE REVIEW

File: connectors/github/element-templates/github-connector.json
Comment (ev-codes) [inline connectors/github/element-templates/github-connector.json:732]: **Note for reviewers:** [API docs](https://docs.github.com/en/rest/actions/workflows?apiVersion=2022-11-28#create-a-workflow-dispatch-event) specify a 10 property maximum.

It appears these kinds of business logic limits usually aren't mentioned or enforced in element templates, but I wanted to flag here just in case.

File: connectors/github/element-templates/github-connector.json
Comment (ev-codes) [inline connectors/github/element-templates/github-connector.json:1229]: [Note to reviewers](https://github.com/camunda/connectors/pull/1634#discussion_r1442975257) that was lost in rebase.

**Comment by ev-codes** [comment]
Even though this PR has an approval, I won't merge it for a few days because:
- It still requires QA work before merging
- I will need some more time to also prepare a docs PR and get the preview environment working again

**Comment by sbuettner** [comment]
@ev-codes You can also use the `dev` and import your Connector into a process model to test it as it does not include any REST Connector code changes.

**Comment by ev-codes** [comment]
@sbuettner thank you!

I just finished testing the Connector in the `dev` environment, and it works as expected!

On Monday, I'll:
- Merge this PR in (unless any objections come up)
- Create a new web-modeler PR adding this OOTB connector version

Sara and I will then use the web-modeler PR for QA work.

**Review by igpetrov — DISMISSED** [review]
LGTM, thank you!

**Comment by ev-codes** [comment]
@sbuettner I updated the connector with the documentation changes [discussed here](https://camunda.slack.com/archives/C05K4TAEPDW/p1705401842225709?thread_ts=1705396558.996289&cid=C05K4TAEPDW).

I tested the changes in the dev environment, and they appear as expected.

If these look ok to you, may you please approve the PR?

Dev environment screenshots:
Screenshot 2024-01-16 at 1 04 33 PM
Screenshot 2024-01-16 at 1 04 49 PM

**Review by sbuettner — APPROVED** [review]

**Pull request merged**
